### PR TITLE
feat: adding in trait chips

### PR DESCRIPTION
**Pull request by aballerr**
1. Added in trait chips
2. Added in trait chips for price range only min, only max and both min and max

### CODE REVIEW

File: src/nft/components/collection/CollectionNfts.tsx [hunk 1]
```diff
@@ -37,6 +42,17 @@ interface CollectionNftsProps {
 
 const rarityStatusCache = new Map<string, boolean>()
 
+const ClearAllButton = styled.button`
+  color: ${({ theme }) => theme.textTertiary};
+  padding-left: 8px;
+  padding-right: 8px;
+  font-size: 14px;
+  font-weight: bold;
```
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:50]: I believe all buttons are supposed to be semibold fontWeight (600)

File: src/nft/components/collection/CollectionNfts.tsx [hunk 2]
```diff
@@ -208,6 +229,16 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
...
 
+  let minMaxPriceChipText = ''
+
+  if (debouncedMinPrice && debouncedMaxPrice) {
+    minMaxPriceChipText = `Price: ${debouncedMinPrice}-${debouncedMaxPrice} ETH`
+  } else if (debouncedMinPrice) {
+    minMaxPriceChipText = `Min. Price: ${debouncedMinPrice} ETH`
+  } else if (debouncedMaxPrice) {
+    minMaxPriceChipText = `Max Price: ${debouncedMaxPrice} ETH`
+  }
```
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:240]: Could do a const and useMemo here

File: src/nft/components/collection/CollectionNfts.tsx [hunk 3]
```diff
@@ -255,6 +286,43 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
...
+              <TraitChip
+                value={minMaxPriceChipText}
+                onClick={() => {
+                  setMin('')
+                  setMax('')
+                }}
+              />
+            )}
+
+            {traits.length || markets.length > 0 ? <ClearAllButton onClick={reset}>Clear All</ClearAllButton> : null}
```
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:324]: Should this check also include if `minMaxPriceChipText` is not empty?

File: src/nft/components/collection/TraitChip.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  border: none;
+`
+
+export const TraitChip = ({ onClick, value }: { value: string; onClick: () => void }) => {
+  return (
+    <TraitChipWrap>
+      <span>{value}</span>
+
+      <CrossIconWrap onClick={onClick}>
+        <CrossIcon style={{ marginTop: 2 }} cursor="pointer" />
```
Comment (cbachmeier) [inline src/nft/components/collection/TraitChip.tsx:28]: Could these styles be added to the wrapper?

File: src/nft/components/collection/TraitChip.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import styled from 'styled-components/macro'
+
+const TraitChipWrap = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 8px 4px 8px 12px;
+  font-weight: 600;
+  border-radius: 12px;
+  background-color: ${({ theme }) => theme.backgroundModule};
```
Comment (cbachmeier) [inline src/nft/components/collection/TraitChip.tsx:11]: should be backgroundAction

File: src/nft/components/collection/CollectionNfts.tsx [hunk 6]
```diff
@@ -255,6 +286,43 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
...
+                }}
+              />
+            ))}
+            {traits.map((trait) => (
+              <TraitChip
+                key={trait.trait_value}
+                value={
+                  trait.trait_type === 'Number of traits'
+                    ? `${trait.trait_value} trait${pluralize(Number(trait.trait_value))}`
+                    : trait.trait_value
```
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:306]: Trait chips are now supposed to have the text include the type of trait for example: `Background: green` currently these chips just have the value, ie `green`

File: src/nft/components/collection/CollectionNfts.tsx [hunk 7]
```diff
@@ -255,6 +286,43 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
...
+                  scrollToTop()
+                  removeTrait(trait)
+                }}
+              />
+            ))}
+            {minMaxPriceChipText && (
+              <TraitChip
+                value={minMaxPriceChipText}
+                onClick={() => {
+                  setMin('')
```
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:318]: missing scrollToTop

File: src/nft/components/collection/CollectionNfts.tsx (hunk 3 above)
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:324]: Should probably also scrollToTop on reset

**Review by cbachmeier — CHANGES_REQUESTED** [review]

File: src/nft/components/collection/TraitChip.tsx (hunk 5 above)
Comment (aballerr) [reply src/nft/components/collection/TraitChip.tsx:11]: I looked at the colors, I found this specific color under ```deprecated_bg3```.  So I made it backgroundModule instead

File: src/nft/components/collection/CollectionNfts.tsx [hunk 8]
```diff
@@ -315,13 +317,22 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
...
                   setMax('')
                 }}
               />
             )}
-
-            {traits.length || markets.length > 0 ? <ClearAllButton onClick={reset}>Clear All</ClearAllButton> : null}
+            {traits.length || markets.length > 0 || minMaxPriceChipText ? (
+              <ClearAllButton
+                onClick={() => {
+                  reset()
```
Comment (cbachmeier) [inline src/nft/components/collection/CollectionNfts.tsx:329]: the reset button doesn't clear min/max price

**Review by cbachmeier — CHANGES_REQUESTED** [review]
looks good, one small fix and I'll be good to approve

**Review by cbachmeier — APPROVED** [review]
LGTM

**Pull request merged**